plugins/tech-insights-backend/migrations/20230213170839_latest-facts-index.js

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const indexName = 'fact_id_entity_timestamp_idx';
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+exports.up = async knex => {
+  await knex.schema.alterTable('facts', table => {
+    table.index(['id', 'entity', 'timestamp'], indexName);
```

Comment:
Should we even make this order by `desc`, since usually we'd want to query more on latest entries?

Reply:
We could, but it's not supported by knex.
https://github.com/knex/knex/issues/5327
We could use raw query, as suggested in linked ticket, but I do not know what is the support in mysql or sqlite for this.

Reply:
Yup, it's grand without as well and likely not that much of a difference